xray/core/dataarray.py

```diff
@@ -233,11 +242,13 @@ def subset(dim, label):
         del coords[dim]
         return Dataset(variables, coords, self.attrs)
 
-    def _to_dataset_whole(self, name):
+    def _to_dataset_whole(self, name=None):
         if name is None:
```

Comment:
Very minor, but `name = name or self.name` might be clearer / more pythonic than the repeated `if name is None:`